### DISCUSSION TITLE
osu!catch Timing-based colouring

### DISCUSSION THREAD

**Discussion by sh0ckR6** [ppy/osu#15314] [Ideas]
Building off of #8583 and #15220, I think adding timing-based colouring to osu!catch would be good just to keep consistency throughout each ruleset as much as possible. I was considering adding osu!taiko to this as well but those colours can't be changed simply due to how the mode works.

##### Not familiar with osu!catch terms so I'm going off of the wiki. Sorry if something sounds wrong here!

---
Fruits/juice drops would be coloured according to their timing. Droplets could also be coloured based on timing, but I don't know how I feel about this as this seems like unnecessary colouring.

I would be more than happy to implement this if this is something that is wanted!

**Comment by peppy** [comment]
I don't see a reason for not propagating this to all modes. The setting may need to be moved to a higher level at some point, though.

**Reply by sh0ckR6** [reply]
For sure, propagating this to all modes at a higher level would be great!

The only issue I have with osu!taiko is that as far as I know, recolouring the object wouldn't work as the distinct colours are required to tell if an object is a don or kat.

If I'm wrong about that or there's a user-friendly way to get around this issue, I think this would be a great addition to osu!taiko.

**Reply by peppy** [reply]
It's on par with the fact that on osu! ruleset, this feature is destroying combo colouring. In both cases we may end up with a solution which avoids this (a new element or something).

**Comment by ekrctb** [comment]
As an osu!catch player I personally don't think that feature is useful (not a strong opinion).
Note that tiny droplets are not snapped to rhythm (and has no hit sound) so it probably shouldn't be colored.

**Reply by sh0ckR6** [reply]
Makes sense. I'm not too familiar with osu!catch as I don't play it very often, so I wasn't sure how useful this would actually be. I was hoping someone who's an osu!catch player would give an opinion on this!

Thinking about it now, it doesn't really make a whole lot of sense to add this to osu!catch as it doesn't seem as timing-based as osu!std or osu!mania. If [this](https://github.com/ppy/osu/discussions/15314#discussioncomment-1542531) ever gets implemented, then osu!catch would fall under this umbrella, but until then I don't see a good reason to actually implement this. It would also just cause more work when the time comes to make it a higher-level setting that gets applied to all rulesets.

##### (Also I had no idea that the tiny droplets weren't snapped to rhythm! Learn something new everyday :D)
